e2e/mock/server.go

```diff
@@ -3,39 +3,78 @@ package mock
 import (
 	"net"
 
-	"github.com/hyperledger/fabric-protos-go/orderer"
-	"github.com/hyperledger/fabric-protos-go/peer"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
 )
 
 type Server struct {
-	GrpcServer *grpc.Server
-	Listener   net.Listener
+	peers   []*Peer
+	orderer *Orderer
 }
 
-func (s *Server) Start() {
+func ConstructServer(peerN int, Credentials credentials.TransportCredentials) (server *Server, err error) {
 	blockC := make(chan struct{}, 1000)
-
-	p := &Peer{
-		BlkSize: 10,
-		TxC:     blockC,
+	server = &Server{}
+	server.peers = make([]*Peer, 0)
+	// for peers
```

Comment:
if a comment is needed, pls put effort in writing a proper/complete sentence - it's for others to read

Reply:
(and i think code here is sufficiently self-explanatory, therefore no comment needed)